pennylane/operation.py

```diff
@@ -1761,10 +1765,43 @@ def matrix(self, wire_order=None):
             # append diagonalizing unitary for specific wire to U_list
             U_list.append(mats[0])
 
+        mat_size = np.prod([np.shape(mat)[0] for mat in U_list])
+        wire_size = 2 ** len(self.wires)
+        if mat_size != wire_size:
+            if partial_overlap:
+                warnings.warn(
+                    "The matrix for Tensors of Tensors/Observables with partially "
```

Comment:
```suggestion
                    "The matrix for tensors of Tensors/Observables with partially "
```
(since as far as I can tell it is referring to the action of taking the tensor product, rather than the `Tensor` class specifically?)

Reply:
No, I think mathematically everything would be well-defined, we just don't implement it here that way. So I would consider it a "feature" of the class :wink: :D